src/pages/Dashboard/Modals/SupplyWithdraw/index.tsx

```diff
@@ -0,0 +1,152 @@
+  const tokenInfo: ILabeledInlineContentProps[] = asset
+    ? [
+        {
+          label: t('supplyWithdraw.supplyApy'),
+          iconName: assetId as IconName,
+          children: formatApy(asset.supplyApy),
+        },
+        {
+          label: t('supplyWithdraw.distributionApy'),
+          iconName: 'xvs' as IconName,
+          children: formatApy(asset.xvsSupplyApy),
+        },
+      ]
+    : [];
+
+  const onSubmit: IAmountFormProps['onSubmit'] = () => {
+    // TODO: https://app.clickup.com/t/24quhp4
+  };
+  const calculateNewSupplyAmount = (amount: BigNumber) => userTotalBorrowLimit.plus(amount);
+  const calculateNewBorrowAmount = (amount: BigNumber) => userTotalBorrowLimit.minus(amount);
+
+  return (
+    <Modal
+      isOpened={!!assetId}
+      handleClose={onClose}
+      title={assetId ? <Token symbol={assetId as TokenId} variant="h4" /> : undefined}
+    >
+      <>
+        <Tabs
+          tabTitles={[t('supplyWithdraw.supply'), t('supplyWithdraw.withdraw')]}
+          onTabChange={setActiveTabIndex}
+          fullWidth
+        />
+        <div className={className} css={styles.container}>
+          <ConnectWallet message={activeTabTextKeys.connect}>
+            {asset ? (
+              <EnableToken
+                symbol={assetId as TokenId}
+                title={activeTabTextKeys.enable}
+                tokenInfo={tokenInfo}
+                isEnabled={!!isEnabled}
+                vtokenAddress={asset.vtokenAddress}
+              >
+                {
+                  [
+                    <SupplyWithdrawForm
+                      key="supply"
+                      asset={asset}
+                      tokenInfo={tokenInfo}
+                      userTotalBorrowBalance={userTotalBorrowBalance}
+                      userTotalBorrowLimit={userTotalBorrowLimit}
+                      dailyEarnings={dailyEarnings}
+                      onSubmit={onSubmit}
+                      enabledButtonKey={t('supplyWithdraw.supply')}
+                      disabledButtonKey={t('supplyWithdraw.enterValidAmountSupply')}
+                      maxInput={asset.walletBalance}
+                      calculateNewBalance={calculateNewSupplyAmount}
+                    />,
+                    <SupplyWithdrawForm
+                      key="withdraw"
+                      asset={asset}
+                      tokenInfo={tokenInfo}
+                      userTotalBorrowBalance={userTotalBorrowBalance}
+                      userTotalBorrowLimit={userTotalBorrowLimit}
+                      dailyEarnings={dailyEarnings}
+                      onSubmit={onSubmit}
+                      enabledButtonKey={t('supplyWithdraw.withdraw')}
+                      disabledButtonKey={t('supplyWithdraw.enterValidAmountWithdraw')}
+                      maxInput={asset.supplyBalance}
+                      calculateNewBalance={calculateNewBorrowAmount}
+                    />,
+                  ][activeTabIndex]
```

Comment:
It took me a minute to understand what the nested array was for ^^
How about just using a ternary operator?
```jsx
{activeTabIndex == 0 ? (
  <SupplyWithdrawForm
    key="supply"
    asset={asset}
    tokenInfo={tokenInfo}
    userTotalBorrowBalance={userTotalBorrowBalance}
    userTotalBorrowLimit={userTotalBorrowLimit}
    dailyEarnings={dailyEarnings}
    onSubmit={onSubmit}
    enabledButtonKey={t('supplyWithdraw.supply')}
    disabledButtonKey={t('supplyWithdraw.enterValidAmountSupply')}
    maxInput={asset.walletBalance}
    calculateNewBalance={calculateNewSupplyAmount}
  />
) : (
  <SupplyWithdrawForm
    key="withdraw"
    asset={asset}
    tokenInfo={tokenInfo}
    userTotalBorrowBalance={userTotalBorrowBalance}
    userTotalBorrowLimit={userTotalBorrowLimit}
    dailyEarnings={dailyEarnings}
    onSubmit={onSubmit}
    enabledButtonKey={t('supplyWithdraw.withdraw')}
    disabledButtonKey={t('supplyWithdraw.enterValidAmountWithdraw')}
    maxInput={asset.supplyBalance}
    calculateNewBalance={calculateNewBorrowAmount}
  />
)}
```

Reply:
I think the array is preferable because it supports more than two tabs. I imagine that when we update the tab to mange its own state and render children that it will use an array

Reply:
Ok got you. Sounds good to me in that case